### PR TITLE
STORY-13556- - Update Github Actions Pipeline

### PR DESCRIPTION
**Pull request by invocandg**
- Update the Github Actions Pipeline of invoca-utils to include Appraisal tests

### CODE REVIEW

File: .gitignore
Comment (invocandg) [inline .gitignore:20]: fyi - This is the only addition, everything else is just organizing the ignores into meaningful chunks

File: .github/workflows/pipeline.yml
Comment (invocandg) [inline .github/workflows/pipeline.yml:14]: fyi - Adding support for `Appraisal` gem files

File: .github/workflows/pipeline.yml
Comment (invocandg) [inline .github/workflows/pipeline.yml:19]: fyi - Setting the excludes as `ActiveSupport@7` requires `Ruby > 2.7`

**Review by retroGiant89 — APPROVED** [review]

File: .github/workflows/pipeline.yml
Comment (jebentier) [inline .github/workflows/pipeline.yml:8]: ❓ is there a reason to change this to a specific version rather than sticking with the latest? This seems to add one more thing our team needs to keep on top of when 22.04 is removed from service down the line.

File: .github/workflows/pipeline.yml
Comment (jebentier) [inline .github/workflows/pipeline.yml:31]: 🔧 We don’t require names for steps that are self evident by the action that they are using. Could we set this back?

File: .github/workflows/pipeline.yml
Comment (jebentier) [inline .github/workflows/pipeline.yml:33]: 🔧 Same request her please

File: CHANGELOG.md
Comment (jebentier) [inline CHANGELOG.md:7]: 🔧 These changes are test only, so there’s no need for a version bump and release. Would you back out the version bump if you haven’t released a version yet please?

**Review by jebentier — COMMENTED** [review]
@invocandg I apologize for the retroactive change requests here once again. I know we’re trying to move fast here and I’ll keep up with these PRs as best I can. To keep the thrash down we might want to wait for owning team approval or limit the changes to smaller change sets where possible.

File: .github/workflows/pipeline.yml
Comment (invocandg) [reply .github/workflows/pipeline.yml:8]: Good point @jebentier 

Addressed the feedback in a follow-up PR at https://github.com/Invoca/invoca-utils/pull/26/files

And to the reason on why I decided to lock to a specific version in the first place is that the [tool](https://github.com/nektos/act) that I was using to run the GitHub actions is failing on `ubuntu-latest` but works fine with `ubuntu-22.04`. Moreover, it was never a good idea to lock to the latest which is unpredictable (as it could change underneath). Analogues to the docker latest tag. But to your feedback and thinking twice that's probably not applicable in this situation.

File: .github/workflows/pipeline.yml
Comment (invocandg) [reply .github/workflows/pipeline.yml:31]: Although we may be a bit succinent in defining actions that are self-evident, I believe it doesn't hurt to be a bit explicit especially when it comes to yaml files that are harder to read/understand the flow. I think this is one of those areas where readability trumps conciseness. Let me know your thoughts.

File: CHANGELOG.md
Comment (invocandg) [reply CHANGELOG.md:7]: Unfortunately, I have already released the gem. Will keep it in mind for future changes.

**Pull request merged**
